scripts/add_link_slope_to_network.py

```diff
@@ -52,15 +57,34 @@
         epsg=projection
     )
 
-    logging.info('Adding elevation to network nodes')
-    n.add_elevation_to_nodes(elevation_tif_file_path=elevation, null_value=tif_null_value)
+    logging.info('Creating elevation dictionary for network nodes')
+    elevation_dictionary = n.get_node_elevation_dictionary(elevation_tif_file_path=elevation, null_value=tif_null_value)
+
+    with open(os.path.join(output_dir, 'elevation/node_elevation_dictionary.json'), 'w', encoding='utf-8') as f:
```

Comment:
this has two problems:

1. we're hardcoding this slash `/` which means a windows user will have problems (more than they usually do anyway); you can do this: `os.path.join(output_dir, 'elevation', 'node_elevation_dictionary.json')` --- the method can join multiple strings into a path

2. more important: we're only calling ensure dir on the `output_dir`, not `output_dir/elevation` so that folder will not exist and the save will fail. I suggest doing:
```
elev_out_dir = os.path.join(output_dir, 'elevation')
ensure_dir(elev_out_dir)
```
and use `elev_out_dir` later on

Reply:
good point!